File: in_Java/0190 Reverse Bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int res = 0;
        int id = 0;
        while(id < 32){
            int end = n & 1;
            n >>=1;
            res <<=1;
            res |= end;
            id++;
        }
        return res;
    }
}

File: in_Python/0190 Reverse Bits.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        res = 0
        p = 0
        while n != 0:
            c = n % 2
            n /= 2
            res += c  * pow(2, 31 - p)
            p += 1
        return res